### DISCUSSION TITLE
Can I issue a redirect via htmx websockets?

### DISCUSSION THREAD

**Discussion by kwint** [bigskysoftware/htmx#3058] [Q&A] [answered]
I would like to tell my clients to go to a different page by sending websocket messages. Is there a way to do this?

I was thinking of sending a div that pulls a url that has the HX-Redirect header, but hopefully there is a cleaner way to do this

**Comment by Telroshan** [comment, marked as answer]
Hey, you could probably do it with a simple event listener, using either [`htmx:wsAfterMessage`](https://htmx.org/extensions/ws/) or [`htmx:wsBeforeMessage`](https://htmx.org/extensions/ws/), and the native [`location.href`](https://developer.mozilla.org/en-US/docs/Web/API/Location/href) or [`location.replace`](https://developer.mozilla.org/en-US/docs/Web/API/Location/replace) depending on your usecase.
Example:
```js
wsSource.addEventListener("htmx:wsAfterMessage", function (event) {
    if (event.detail.message === "redirect") {
        document.location.href = "/my/url"
    }
})
```
Or you could for example define a custom syntax like a `redirect:` prefix then the target URL, for your JS to parse, or send a JSON with the target URL... Likely many ways to implement it!

Hope this helps!

**Reply by kwint** [reply]
Thats perfect, thanks a lot!
